apps/ogr2ogr_lib.cpp

```diff
@@ -6207,49 +6228,56 @@ bool LayerTranslator::Translate(
 
                 if (m_poClipSrcOri)
                 {
+                    if (poDstGeometry->IsEmpty())
+                        goto end_loop;
 
-                    const OGRGeometry *poClipGeom =
+                    auto [poClipGeom, poClipGeomEnvelope] =
                         GetSrcClipGeom(poDstGeometry->getSpatialReference());
 
-                    std::unique_ptr<OGRGeometry> poClipped;
-                    if (poClipGeom != nullptr)
-                    {
-                        OGREnvelope oClipEnv;
-                        OGREnvelope oDstEnv;
+                    if (!(poClipGeom && poClipGeomEnvelope))
+                        goto end_loop;
 
-                        poClipGeom->getEnvelope(&oClipEnv);
-                        poDstGeometry->getEnvelope(&oDstEnv);
+                    OGREnvelope oDstEnv;
+                    poDstGeometry->getEnvelope(&oDstEnv);
 
-                        if (oClipEnv.Intersects(oDstEnv))
+                    bool bContained = false;
```

Comment:
can't you just remove bContained and move the check to line 6248?

Reply:
done